### PR TITLE
Added length validation for [p]set name and [p]set nickname

### LINKED ISSUE #4363: [Core] Set name and nickname should handle length errors in a user friendly way
# Command bugs
#### Command name
`[p]set name` and `[p]set nickname`
#### What cog is this command from?
Core
#### What were you expecting to happen?
The bot to friendly say the name was too long
#### What actually happened?
`[p]set name` sent "Failed to change name. Remember that you can only do it up to 2 times an hour.", which is an unrelated error message
`[p]set nickname` fully errored
#### How can we reproduce this issue?
Use the commands above with a really long name

### PR DESCRIPTION
**Pull request by hisztendahl**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Return error message if [p]set name or [p]set nickname string is too long

Fixes #4363

### CODE REVIEW

File: redbot/core/core_commands.py [hunk 1]
```diff
@@ -1919,7 +1919,12 @@ async def stream(self, ctx: commands.Context, streamer=None, *, stream_title=Non
     async def _username(self, ctx: commands.Context, *, username: str):
         """Sets [botname]'s username."""
         try:
-            await self._name(name=username)
+            if len(username) > 32:
+                await ctx.send(_("Failed to change name. Must be 32 characters or fewer."))
+                return
+            else:
+                print("USERNAME LENGTH OK")
```
Comment (TrustyJAID) [inline redbot/core/core_commands.py]: We don't use `print` statements for this. So this should either be removed or changed to `log.debug` if you want to keep it although I don't really see a need for it beyond testing the function worked.

**Review by TrustyJAID — CHANGES_REQUESTED** [review]

File: redbot/core/core_commands.py (hunk 1 above)
Comment (hisztendahl) [reply redbot/core/core_commands.py]: Woops good catch, forgot to remove it

**Review by TrustyJAID — DISMISSED** [review]
Looks good to me!

**Review by Dav-Git — DISMISSED** [review]

**Review by NeuroAssassin — DISMISSED** [review]
Thanks a lot!

**Review by madebylydia — CHANGES_REQUESTED** [review]
Can you remove the else statement? They're not a real necessity in here.

**Review by madebylydia — APPROVED** [review]

**Review by Drapersniper — APPROVED** [review]
Reapproved Trusty's approval after the change Predeactor suggested.

**Pull request merged**